internal/services/task_status_listener.go

```diff
@@ -121,8 +125,16 @@ func (i *TaskStatusListener) processSmartcarPollStatusEvent(event *shared.CloudE
 	}
 
 	udai.Status = models.UserDeviceAPIIntegrationStatusAuthenticationFailure
-	_, err = udai.Update(context.Background(), i.db().Writer, boil.Infer())
-	return err
+	if _, err = udai.Update(context.Background(), i.db().Writer, boil.Infer()); err != nil {
+		i.log.Err(err).Str("userDeviceID", userDeviceID).Str("integrationID", integrationID).Msg("failed up update user device api integration with failure status")
+	}
+
+	if err := i.cioSvc.SoftwareDisconnectionEvent(ctx, udai); err != nil {
+		i.log.Err(err).Str("userDeviceID", userDeviceID).Str("integrationID", integrationID).Msg("failed to send CIO software disconnection event")
```

Comment:
I think this error will be logged by the caller so we don't need to log here as well.